src/rust/engine/src/nodes.rs

```diff
@@ -494,6 +494,15 @@ impl From<Scandir> for NodeKey {
   }
 }
 
+fn unmatched_globs_additional_context() -> Option<String> {
+  Some(format!(
+    "\n\nDoes the file exist? If so, check if the file is in your `.gitignore` or the global \
+    `pants_ignore` option, which may result in Pants not being able to see the file even though it \
+    exists on disk. Refer to {}.",
```

Comment:
@benjyw no need to review the rest of the PR, but you may have thoughts on the wording here.